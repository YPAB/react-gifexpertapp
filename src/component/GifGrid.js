import React from "react";
import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { getGifs } from "../helpers/getGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({category}) => {

    // const [images,setImages] = useState ([]);

    // es para que se renderice una sola vez (Solo se ejecuta cuando el componente es iniciado por primera vez)
   /*  useEffect(() => {
        getGifs(category)
        .then(setImages);
    }, [category])
*/
    //getGifs();

    const {data:images,loading} = useFetchGifs(category);

    return (
        <>

            <h3>{category}</h3>

            <div className='card-grid'>
                {loading && <p>Cargando ...</p>}

                
                    {
                        images.map (img => (

                            <GifGridItem 
                            key={img.id}
                            {...img}/>

                            
                        ))
                    }
                
            </div>
        </>
    )

}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}