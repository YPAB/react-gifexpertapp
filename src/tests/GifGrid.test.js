import React from "react";
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifGrid } from "../component/GifGrid";
import { useFetchGifs } from "../hooks/useFetchGifs";

//uso de mock para falsear data
jest.mock('../hooks/useFetchGifs')

describe ('Prueba con el componente <GifGrid/>>', () => {

    const category = 'Los Simpson';
    
    

    test('Debe crear una copia', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true

        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagens useFectGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa',

        },
        {
            id: '123',
            url: 'https://localhost/cualquier/foto.jpg',
            title: 'Foto',

        }
    
    ];

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false

        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect( wrapper ).toMatchSnapshot();

        // cuando el loading es FALSE se oculta el parrafo de Cargando...
        expect( wrapper.find('p').exists() ).toBe(false);

        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );


    });


})

