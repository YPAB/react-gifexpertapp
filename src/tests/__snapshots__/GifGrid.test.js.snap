// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con el componente <GifGrid/>> Debe crear una copia 1`] = `
<Fragment>
  <h3>
    Los Simpson
  </h3>
  <div
    className="card-grid"
  >
    <p>
      Cargando ...
    </p>
  </div>
</Fragment>
`;

exports[`Prueba con el componente <GifGrid/>> debe de mostrar items cuando se cargan imagens useFectGifs 1`] = `
<Fragment>
  <h3>
    Los Simpson
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Foto"
      url="https://localhost/cualquier/foto.jpg"
    />
  </div>
</Fragment>
`;
